### PR TITLE
Feature/error boundaries

### PR DESCRIPTION
**Pull request by HuyAms**
-  feat: add ErrorBoundaries to handle unexpected error, disable strict mode and noImplicitAny in tsconfig
- refactor: move component to its own directory
- refactor: capitalize component directory for naming convention

### CODE REVIEW

File: src/shared/components/ErrorBoundaries/ErrorBoundaries.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+			errorInfo,
+		})
+	}
+
+	render() {
+		if (this.state.errorInfo) {
+			return (
+				<div className="error-boundaries">
+					<h2 className="error-boundaries__header">Something went wrong</h2>
+					<details style={{ whiteSpace: 'pre-wrap' }}>
```
Comment (vinhlee95) [inline src/shared/components/ErrorBoundaries/ErrorBoundaries.tsx]: Should probably not use inline-styling

File: src/shared/components/ErrorBoundaries/ErrorBoundaries.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+						{this.state.error && this.state.error.toString()}
+						<br />
+						{this.state.errorInfo.componentStack}
+					</details>
```
Comment (vinhlee95) [inline src/shared/components/ErrorBoundaries/ErrorBoundaries.tsx]: IE & Ms.Edge do not support this tag. So let's replace it with another tag
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#Browser_compatibility

Comment (HuyAms) [reply src/shared/components/ErrorBoundaries/ErrorBoundaries.tsx]: It is not fully supported but it works ok in Edge. Edge just treats it as a <p/> tag or something

**Comment by vinhlee95** [comment]
There are conflicts in `CoreLayout`. Please fix it before merging

**Pull request merged**